example/src/main/java/io/netty/example/util/ServerUtil.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.example.util;
+
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+/**
+ * some useful methods for server side.
+ */
+public final class ServerUtil {
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(ServerUtil.class);
+
+    private static final boolean SSL = System.getProperty("ssl") != null;
+
+    private ServerUtil() {
+    }
+
+    public static SslContext buildSslContext() {
+        if (!SSL) {
+            return null;
+        }
+
+        SslContext sslCtx = null;
+        try {
+            SelfSignedCertificate ssc = new SelfSignedCertificate();
+            sslCtx = SslContextBuilder.forServer(ssc.certificate(), ssc.privateKey())
+                    .build();
+        } catch (Exception ex) {
+            logger.warn("exception occurs when get SslContext.", ex);
```

Comment:
Just let the exception bubble out like the examples did previously. No need for any logging as the stack trace will be printed when the exception unwinds `main()`.

Reply:
Thanks for your suggestion.